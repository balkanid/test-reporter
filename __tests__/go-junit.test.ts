import * as fs from 'fs'
import * as path from 'path'

import {GoJunitParser} from '../src/parsers/go-junit/go-junit-parser'
import {ParseOptions} from '../src/test-parser'
import {getReport} from '../src/report/get-report'
import {normalizeFilePath} from '../src/utils/path-utils'

describe('go-junit tests', () => {
  it('produces empty test run result when there are no test cases', async () => {
    const fixturePath = path.join(__dirname, 'fixtures', 'empty', 'jest-junit.xml')
    const filePath = normalizeFilePath(path.relative(__dirname, fixturePath))
    const fileContent = fs.readFileSync(fixturePath, {encoding: 'utf8'})

    const opts: ParseOptions = {
      parseErrors: true,
      trackedFiles: []
    }

    const parser = new GoJunitParser(opts)
    const result = await parser.parse(filePath, fileContent)
    expect(result.tests).toBe(0)
    expect(result.result).toBe('success')
  })

  it('report from ./external/go-junit-report test results matches snapshot', async () => {
    const fixturePath = path.join(__dirname, 'fixtures', 'external', 'go', 'go-junit-report.xml')
    const outputPath = path.join(__dirname, '__outputs__', 'go-junit-report-test-results.md')
    const filePath = normalizeFilePath(path.relative(__dirname, fixturePath))
    const fileContent = fs.readFileSync(fixturePath, {encoding: 'utf8'})

    const opts: ParseOptions = {
      parseErrors: true,
      trackedFiles: ['__tests__/main.test.js', '__tests__/second.test.js', 'lib/main.js']
      //workDir: 'C:/Users/Michal/Workspace/dorny/test-check/reports/jest/'
    }

    const parser = new GoJunitParser(opts)
    const result = await parser.parse(filePath, fileContent)
    expect(result).toMatchSnapshot()

    const report = getReport([result])
    fs.mkdirSync(path.dirname(outputPath), {recursive: true})
    fs.writeFileSync(outputPath, report)
  })

  it('report from ./external/ginkgo test results matches snapshot', async () => {
    const fixturePath = path.join(__dirname, 'fixtures', 'external', 'go', 'ginkgo-junit-report.xml')
    const outputPath = path.join(__dirname, '__outputs__', 'ginkgo-junit-test-results.md')
    const filePath = normalizeFilePath(path.relative(__dirname, fixturePath))
    const fileContent = fs.readFileSync(fixturePath, {encoding: 'utf8'})

    const opts: ParseOptions = {
      parseErrors: true,
      trackedFiles: ['__tests__/main.test.js', '__tests__/second.test.js', 'lib/main.js']
      //workDir: 'C:/Users/Michal/Workspace/dorny/test-check/reports/jest/'
    }

    const parser = new GoJunitParser(opts)
    const result = await parser.parse(filePath, fileContent)
    expect(result).toMatchSnapshot()

    const report = getReport([result])
    fs.mkdirSync(path.dirname(outputPath), {recursive: true})
    fs.writeFileSync(outputPath, report)
  })
})
