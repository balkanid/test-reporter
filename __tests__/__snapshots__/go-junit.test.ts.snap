// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`go-junit tests report from ./external/ginkgo test results matches snapshot 1`] = `
TestRunResult {
  "path": "fixtures/external/go/ginkgo-junit-report.xml",
  "suites": Array [
    TestSuiteResult {
      "groups": Array [
        TestGroupResult {
          "name": "Suite1",
          "tests": Array [
            TestCaseResult {
              "error": undefined,
              "name": "[It] Should succeed",
              "result": "success",
              "time": 0.455833,
            },
          ],
        },
      ],
      "name": "Suite1",
      "totalTime": 1.7596669999999999,
    },
    TestSuiteResult {
      "groups": Array [
        TestGroupResult {
          "name": "Suite2",
          "tests": Array [
            TestCaseResult {
              "error": undefined,
              "name": "[It] Should succeed",
              "result": "success",
              "time": 104.59475,
            },
            TestCaseResult {
              "error": Object {
                "details": "/package2/a_test.go:52",
                "line": undefined,
                "path": undefined,
              },
              "name": "[It] Should succeed, but errors",
              "result": "failed",
              "time": 0.523208,
            },
          ],
        },
      ],
      "name": "Suite2",
      "totalTime": 135.371083,
    },
  ],
  "totalTime": 9568.500872999997,
}
`;

exports[`go-junit tests report from ./external/go-junit-report test results matches snapshot 1`] = `
TestRunResult {
  "path": "fixtures/external/go/go-junit-report.xml",
  "suites": Array [
    TestSuiteResult {
      "groups": Array [],
      "name": "github.com/company/package1",
      "totalTime": 0,
    },
    TestSuiteResult {
      "groups": Array [
        TestGroupResult {
          "name": "github.com/company/package2",
          "tests": Array [
            TestCaseResult {
              "error": undefined,
              "name": "TestPackage2",
              "result": "success",
              "time": 0,
            },
          ],
        },
      ],
      "name": "github.com/company/package2",
      "totalTime": 482,
    },
    TestSuiteResult {
      "groups": Array [
        TestGroupResult {
          "name": "github.com/company/package3",
          "tests": Array [
            TestCaseResult {
              "error": Object {
                "details": "Just error here",
                "line": undefined,
                "path": undefined,
              },
              "name": "TestPackage3-1",
              "result": "failed",
              "time": 0,
            },
            TestCaseResult {
              "error": undefined,
              "name": "TestPackage3-2",
              "result": "success",
              "time": 160,
            },
            TestCaseResult {
              "error": undefined,
              "name": "TestPackage3-3",
              "result": "skipped",
              "time": 0,
            },
            TestCaseResult {
              "error": Object {
                "details": "",
                "line": undefined,
                "path": undefined,
              },
              "name": "TestPackage3-4",
              "result": "failed",
              "time": 0,
            },
          ],
        },
      ],
      "name": "github.com/company/package3",
      "totalTime": 835,
    },
  ],
  "totalTime": undefined,
}
`;
